.gitlab-ci.yml

```diff
@@ -57,3 +60,60 @@ deploy-images-dev:
   artifacts:
     reports:
       dotenv: build.env
+
+e2e-tests-dev:
+  stage: e2e-tests-dev
+  # This resource group is configured with process_mode=oldest_first to make sure the pipelines are run serially.
+  resource_group: production
+  image: $CNS_IMAGE_GOLANG
+  dependencies:
+    - deploy-images-dev
+  script:
+    - ./pipeline/e2e-tests.sh
+
+deploy-images-staging:
+  stage: deploy-staging
+  # Image built from cd-infra/images/ci-deploy/Dockerfile from Calatrava project.
+  image: $CNS_IMAGE_CI_DEPLOY_STAGE
+  script:
+    - ./pipeline/deploy-staging.sh
+  dependencies:
+    - build-images
+  only:
+    - master
```

Comment:
Does this mean e2e tests in test environment will run from any branch, but staging deployment and tests will only run for commits to master branch?

Reply:
Yes.